src/systems/joint_trajectory_controller/JointTrajectoryController.hh

```diff
@@ -0,0 +1,132 @@
+#ifndef IGNITION_GAZEBO_SYSTEMS_JOINT_TRAJECTORY_CONTROLLER_HH_
+#define IGNITION_GAZEBO_SYSTEMS_JOINT_TRAJECTORY_CONTROLLER_HH_
+
+#include <ignition/gazebo/System.hh>
+
+namespace ignition
+{
+  namespace gazebo
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE
+    {
+      namespace systems
+      {
+        // Forward declaration
+        class IGNITION_GAZEBO_HIDDEN JointTrajectoryControllerPrivate;
+
+        /// \brief Joint trajectory controller, which can be attached to a model with
+        /// reference to one or more 1-axis joints in order to follow a trajectory.
+        ///
+        /// Joint trajectories can be sent to this plugin via Ignition Transport.
+        /// The default topic name is "/model/${MODEL_NAME}/joint_trajectory".
+        /// This topic name can be configured with the `<topic>` system parameter.
+        ///
+        /// This topic accepts ignition::msgs::JointTrajectory messages that represent
+        /// a full trajectory, defined as temporal `points` with their fields ordered
+        /// according to `joint_names` field. The fields under `points` are
+        /// `positions` - Controlled by position PID controller for each joint
+        /// `velocities` - Controlled by velocity PID controller for each joint
+        /// `accelerations` - This field is currently ignored
+        /// `effort` - Controlled directly for each joint
+        /// `time_from_start` - Temporal information about the point
+        ///
+        /// Forces/torques from `position`, `velocity` and `effort` are summed and
+        /// applied to the joint. Each PID controller can be configured with system
+        /// parameters described below.
+        ///
+        /// This trajectory can be produced by a motion planning framework, e.g. MoveIt2.
+        ///
+        /// The progress of the current trajectory can be similarly tracked on
+        /// another topic whose name is derived as `<topic>_progress`.
```

Comment:
Using a service here instead of pub+sub could potentially be better. However, interoperability with ROS 2 does not allow this as [ros_ign_bridge](https://github.com/ignitionrobotics/ros_ign/tree/ros2/ros_ign_bridge) has no support for service calls at the moment. Hope that's fine.

Reply:
Yeah I think the best that can be done right now is using pub/sub. If https://github.com/ignitionrobotics/ros_ign/issues/70 is ever addressed, an extra service interface could be added to this plugin.